audioswitch/src/main/java/com/twilio/audioswitch/AudioSwitch.kt

```diff
@@ -105,21 +103,36 @@ class AudioSwitch {
      *
      * @param context An Android Context.
      * @param loggingEnabled Toggle whether logging is enabled. This argument is false by default.
-     * @param audioFocusChangeListener A listener that is invoked when the system audio focus is
-     * updated. Note that updates are only sent to the listener after [activate] has been called.
+     * @param audioFocusChangeListener A listener that is invoked when the system audio focus is updated.
+     * Note that updates are only sent to the listener after [activate] has been called.
+     * @param preferredDeviceList The order in which [AudioSwitch] automatically selects and activates
+     * an [AudioDevice]. This parameter is ignored if the [selectedAudioDevice] is not null.
+     *
+     * The default preferred [AudioDevice] order is the following:
+     *
+     * [BluetoothHeadset], [WiredHeadset], [Earpiece], [Speakerphone]
+     *
+     * [preferredDeviceList] is added to the front of the default list. For example, if [preferredDeviceList]
+     * is [Speakerphone] and [BluetoothHeadset], then the new preferred audio
+     * device list will be:
+     *
+     * [Speakerphone], [BluetoothHeadset], [WiredHeadset], [Earpiece].
      */
     @JvmOverloads
     constructor(
         context: Context,
         loggingEnabled: Boolean = false,
-        audioFocusChangeListener: OnAudioFocusChangeListener = OnAudioFocusChangeListener {}
-    ) : this(context.applicationContext, Logger(loggingEnabled), audioFocusChangeListener)
+        audioFocusChangeListener: OnAudioFocusChangeListener = OnAudioFocusChangeListener {},
+        preferredDeviceList: List<Class<out AudioDevice>> = defaultPreferredDeviceList
+    ) : this(context.applicationContext, ProductionLogger(loggingEnabled), audioFocusChangeListener,
+            preferredDeviceList)
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal constructor(
         context: Context,
         logger: Logger,
         audioFocusChangeListener: OnAudioFocusChangeListener,
+        preferredDeviceList: List<Class<out AudioDevice>>,
```

Comment:
The the `out AudioDevice` indicate that only instances of an `AudioDevice` class can be provided?

Reply:
Yep! It guarantees that only classes that extend AudioDevice are allowed.